java/java.source.base/src/org/netbeans/modules/java/source/parsing/VanillaPartialReparser.java

```diff
@@ -0,0 +1,425 @@
+                        start = System.currentTimeMillis();
+                        reattrMethodBody(ctx, methodScope, orig, block);
+                        if (LOGGER.isLoggable(Level.FINER)) {
+                            LOGGER.log(Level.FINER, "Resolved method in: {0}", fo);     //NOI18N
+                        }
+                        if (!((CompilationInfoImpl.DiagnosticListenerImpl)dl).hasPartialReparseErrors()) {
+                            final JavacFlowListener fl = JavacFlowListener.instance(ctx);
+                            if (fl != null && fl.hasFlowCompleted(fo)) {
+                                if (LOGGER.isLoggable(Level.FINER)) {
+                                    final List<? extends Diagnostic> diag = ci.getDiagnostics();
+                                    if (!diag.isEmpty()) {
+                                        LOGGER.log(Level.FINER, "Reflow with errors: {0} {1}", new Object[]{fo, diag});     //NOI18N
+                                    }
+                                }
+                                TreePath tp = TreePath.getPath(cu, orig);       //todo: store treepath in changed method => improve speed
+                                Tree t = tp.getParentPath().getLeaf();
+                                reflowMethodBody(ctx, cu, (ClassTree) t, orig);
+                                if (LOGGER.isLoggable(Level.FINER)) {
+                                    LOGGER.log(Level.FINER, "Reflowed method in: {0}", fo); //NOI18N
+                                }
+                            }
+                        }
+                        end = System.currentTimeMillis();
+                        if (fo != null) {
+                            JavacParser.logTime (fo, Phase.ELEMENTS_RESOLVED,0L);
+                            JavacParser.logTime (fo,Phase.RESOLVED,(end-start));
+                        }
+                    }
+
+                    //fix CompilationUnitTree.getLineMap:
+                    long startM = System.currentTimeMillis();
+                    char[] chars = snapshot.getText().toString().toCharArray();
+                    if (lineMapBuild != null) {
+                        lineMapBuild.invoke(cu.getLineMap(), chars, chars.length);
+                    }
+                    LOGGER.log(Level.FINER, "Rebuilding LineMap took: {0}", System.currentTimeMillis() - startM);
+
+                    ((CompilationInfoImpl.DiagnosticListenerImpl)dl).endPartialReparse (delta);
+                } finally {
+                    l.endPartialReparse(cu.getSourceFile());
+                    l.useSource(prevLogged);
+                }
+                ci.update(snapshot);
+            } finally {
+//              if (treeLoader != null) {
+//                  treeLoader.endPartialReparse();
+//              }
+            }
+//        } catch (CouplingAbort ca) {
+//            //Needs full reparse
+//            return false;
+        } catch (Throwable t) {
```

Comment:
Jackpot:
warning: Can be replaced with multicatch or several catch clauses catching specific exceptions